checkstyle/src/main/java/org/infinispan/checkstyle/checks/ForbiddenMethodCheck.java

```diff
@@ -0,0 +1,106 @@
+package org.infinispan.checkstyle.checks;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Checks that certain methods are not invoked
+ *
+ * The methodPattern argument is a regular expression, invocations of method names matching the pattern are forbidden.
+ * Only method names are checked, the defining class doesn't matter.
+ *
+ * If the optional argumentCount argument is set, only overloads with that number of parameters are forbidden.
+ * Forbidding only overloads with 0 arguments is not supported at this time.
+ *
+ * Optionally can check test invocations if {@link #setCheckTestClasses(String)} is invoked with true. Default is false.
+ *
+ * @author wburns
+ * @since 10.0
+ */
+
+public class ForbiddenMethodCheck extends AbstractCheck {
+   private Pattern methodPattern = null;
+   private int argumentCount = -1;
+   private boolean checkTestClasses = false;
```

Comment:
I don't see this actually being used, and the plugin's `includeTestSourceDirectory` config already handles this.

Reply:
Is used at https://github.com/infinispan/infinispan/pull/7108/files#diff-5cf69a433c2c6fce8c5a1f248163cf12R101 

Also `includeTestSourceDirectory` is at the entire checkstyle level if I understand it correctly. You may want to have some assertions be for tests and others to not individually I would think.

Reply:
AFAICT that `if` just changes the log message, not whether the check is active or not.

And maybe for other checks you really want them to be active only in production code, but

* using the common pool in tests makes it harder to read logs, so it should be forbidden
* any difference between prod and test checkstyle rules is one more thing to remember, and I'd rather spend my limited memory capacity on other things :)

Reply:
Yeah it changes the log message which then determines if it is filtered or not as we use the `ExcludeTestPackages` which looks for that. I figured that would be how to do it, if not we should probably remove that class then.

Reply:
I can refactor the test uses and remove the variable - I am not that picky either way. Just figured it sometimes make tests simpler. But as you mention thread leaks are harder.

Reply:
I see, I had seen `ExcludeTestPackages` in the checkstyle config before but I had no idea how it works.

I see the interceptor checks rely on `ExcludeTestPackages`, so we can't remove it or we'd drown in false positives. OTOH I don't think it's needed for this new check, and I'd rather the interceptor checks use a marker interface or something else to distinguish test interceptors. See also https://issues.jboss.org/browse/ISPN-10150